cfme/tests/middleware/test_middleware_datasource.py

```diff
@@ -14,6 +16,13 @@
 ITEMS_LIMIT = 2  # when we have big list, limit number of items to test
 
 
+@pytest.yield_fixture(scope="function")
+def oracle_database():
```

Comment:
if there is always a need to ensure allocation/deallocation,

how about making that db_allocate a context-manager to begin with

```
with allocated_db(ORACLE_12C) as db_metadata:
  yield db_metadata
```
seems like a potential candidate